docs/pages/nexus-client/methods/upgradeSmartAccount.md

```diff
@@ -0,0 +1,164 @@
+- `initData` `Hex` (optional): Initialization data to pass to the new implementation. Default is "0x".
+- `maxFeePerGas` `bigint` (optional): Maximum fee per gas for User Operation execution.
+- `maxPriorityFeePerGas` `bigint` (optional): Maximum priority fee per gas.
+- `nonce` `bigint` (optional): Nonce for the User Operation.
+- Other standard User Operation parameters are also supported.
+
+## Response
+
+- `Promise<Hash>`: Returns the User Operation hash of the upgrade transaction.
+
+## Updating the AbstractJS Package
+
+:::warning[Important]
+Before upgrading smart accounts, you should update the AbstractJS package to the latest version using your package manager. This ensures you have access to the latest implementation addresses and features.
+:::
+
+:::code-group
+```bash [npm]
+npm update @biconomy/abstractjs
+```
+```bash [yarn]
+yarn upgrade @biconomy/abstractjs
+```
+```bash [pnpm]
+pnpm update @biconomy/abstractjs
+```
+```bash [bun]
+bun update @biconomy/abstractjs
+```
+:::
+
+After updating the package, you should also update your application code to use any new APIs or handle any breaking changes in the latest version.
+
+## Migrating from Older Versions
+
+### Using the `oldVersion` Parameter
+
+When migrating from an older version of a Nexus smart account, the `oldVersion` parameter in `toNexusAccount` is crucial for connecting to accounts created with previous implementations.
+
+:::tip[Important for Migrations]
+You must specify the `oldVersion` parameter when initializing your account to ensure you can access the correct account address and funds from older versions.
+:::
+
+```typescript [migration-example.ts]
+// Connect to an older version of the smart account
+const nexusClient = createSmartAccountClient({
+  account: await toNexusAccount({ 
+    signer: account, 
+    chain: baseSepolia,
+    transport: http(),
+    oldVersion: "0.0.33", // Specify the SDK version your users account was created with. Remember to store this in your application to access the upgraded account & balances moving forward
+  }),
+  transport: http(bundlerUrl),
+});
+
+// Now upgrade to the latest version
+const upgradeHash = await nexusClient.upgradeSmartAccount();
+```
+
+## Migration Process
+
+1. **Update AbstractJS package**: Use your package manager to update to the latest version: `npm update @biconomy/abstractjs`
+2. **Connect to existing account**: Use the `oldVersion` parameter to connect to the old account
+3. **Upgrade implementation**: Call `upgradeSmartAccount()` to update it to the latest version
+4. **Verify upgrade**: Test that the account works correctly after upgrading
+5. **Update application**: Modify any code that depends on specific implementation details
+6. **Store the version**: Store the users oldVersion in your application to access the upgraded account & balances moving forward
+
+## Advanced Usage
```

Comment:
I think an example of `initData` may be useful

Also I would rename it to `callData`
Because technically it is not initialization, as the storage is not cleaned, all the modules are kept installed and configured
It is just an opportunity to execute an additional call directly after the upgrade

for example install an additional module